src/main/java/com/redhat/devtools/lsp4ij/features/semanticTokens/viewProvider/LSPSemanticToken.java

```diff
@@ -0,0 +1,260 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.features.semanticTokens.viewProvider;
+
+import com.intellij.openapi.util.TextRange;
+import com.intellij.psi.PsiFile;
+import com.intellij.util.ThreeState;
+import com.intellij.util.containers.ContainerUtil;
+import org.eclipse.lsp4j.SemanticTokenModifiers;
+import org.eclipse.lsp4j.SemanticTokenTypes;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Represents a concrete semantic token in a file.
+ */
+class LSPSemanticToken {
+    // TODO: Need a mechanism for language server definitions to register token types and token modifiers including how
+    //  they should be interpreted for syntax highlighting and categorization
+    private static final String MEMBER_TOKEN_TYPE = "member"; // JavaScript/TypeScript-specific
```

Comment:
This is obviously hard-coded for now or JavaScript/TypeScript simply wouldn't work very well. It's the same token type we discussed previously, and I agree that there needs to be a first-class way for a language server to register semantic token types and modifiers and the way they should be interpreted by LSP4IJ. I'd like to propose that this hard-coded exception be allowed right now and once all of this work is merged, I'll circle back and add a way for language server definitions -- custom and user-defined -- to add new semantic token types and modifiers to LSP4IJ.